### PR TITLE
add @storybook/polymer

### PR DESCRIPTION
**Pull request by ryanis-online**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.) (cannot get past npm naming issue. will open issue)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present). (cannot get this to run without npm naming issue. will open issue)

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/storybook__polymer/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-unnecessary-generics": false
```
Comment (andrewbranch) [inline types/storybook__polymer/tslint.json]: There shouldn’t be anything in `rules`

File: types/storybook__polymer/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// Type definitions for @storybook/polymer 5.1
+// Project: https://github.com/storybookjs/storybook, https://github.com/storybookjs/storybook/tree/master/app/polymer
+// Definitions by: Joscha Feth <https://github.com/joscha>
+//                 Anton Izmailov <https://github.com/wapgear>
+//                 Dan Dean <https://github.com/dandean>
+//                 Ryan Coleman <https://github.com/ryuhhnn>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.4
```
Comment (andrewbranch) [inline types/storybook__polymer/index.d.ts]: At a glance I don't think there's anything in here that requires TypeScript 3.4? This number should be the lowest possible version of TypeScript that is compatible with these definitions.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: types/storybook__polymer/tslint.json (hunk 1 above)
Comment (ryuhhnn) [reply types/storybook__polymer/tslint.json]: Don't know if this is documented anywhere, but there are a large number of packages with tslint rules. I removed them though

**Comment by ryuhhnn** [comment]
@andrewbranch feedback addressed, feel free to ping me if this requires further changes

File: types/storybook__polymer/tslint.json (hunk 1 above)
Comment (andrewbranch) [reply types/storybook__polymer/tslint.json]: It’s here: https://github.com/DefinitelyTyped/DefinitelyTyped#lint

At the time we started linting DefinitelyTyped, there were already thousands of packages. Fixing them all to conform would have been a herculean effort with a questionable rate of return, so we auto-generated tslint configs disabling the smallest number of rules possible for each package. That‘s why you see existing exceptions—in a perfect world, there would be none.

**Comment by andrewbranch** [comment]
It looks like `lit-element` uses the `unknown` type, which is available starting in TypeScript 3.0—that should be the TypeScript version specified in your header.

**Comment by andrewbranch** [comment]
Ugh

```
Error in storybook__polymer
Error: Errors in typescript@3.0 for external dependencies:
node_modules/lit-html/lib/parts.d.ts(16,89): error TS2304: Cannot find name 'bigint'.
```

`bigint` was added in 3.2... sorry Ryan 🤦‍♂

**Review by andrewbranch — APPROVED** [review]

**Pull request merged**